File: movie-domain/domain/src/main/java/com/movie/domain/movie/MovieService.java
package com.movie.domain.movie;

import com.movie.domain.movie.dto.info.MovieInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class MovieService {

//    private final MovieRepository movieRepository;

    public List<MovieInfo.Get> getMovies() {
        return null;
    }
}

File: movie/core/src/main/kotlin/com/example/movie/service/MovieService.kt
package com.example.movie.service

import com.example.movie.domain.common.TimeHandler
import com.example.movie.domain.movie.model.Movie
import com.example.movie.domain.movie.repository.MovieRepository
import com.example.movie.domain.screening.model.ScreeningStatus
import org.springframework.stereotype.Service

@Service
class MovieService(
    private val movieRepository: MovieRepository,
    private val timeHandler: TimeHandler
) : MovieSearchUseCase {

    override fun getMoviesByStatusAndTitleAndGenre(status: ScreeningStatus, title: String?, genreId: Long?): List<Movie> {
        val currentTime = timeHandler.getCurrentTime()
        return movieRepository.findAllByStatusWithMovieAndTheater(currentTime,ScreeningStatus.SCHEDULED, title, genreId)
    }
}